crates/typed-store/src/rocks/iter.rs

```diff
@@ -49,6 +77,8 @@ impl<'a, K: DeserializeOwned, V: DeserializeOwned> Iterator for Iter<'a, K, V> {
                 .db_iter
                 .value()
                 .expect("Valid iterator failed to get value");
+            self.bytes_scanned_counter += raw_key.len() + raw_value.len();
+            self.keys_returned_counter += 1;
```

Comment:
Probably for future work, is there a way to count the deletion tombstones skipped over from the underlying rocksdb iterator, maybe with a sampling interval?

Reply:
oh good point @mwtian, I wanted to add that too. There is a perf ctx metric for that. Let me add that in another PR (otherewise this keeps getting bigger)